src/rust/ensogl/lib/components/src/selector/common/model.rs

```diff
@@ -0,0 +1,282 @@
+//! Base model for the number and range selector components.
+use crate::prelude::*;
+
+use crate::component;
+use crate::selector::common;
+
+use enso_frp as frp;
+use ensogl_core::application::Application;
+use ensogl_core::application;
+use ensogl_core::data::color;
+use ensogl_core::display::shape::*;
+use ensogl_core::display;
+use ensogl_text as text;
+
+use super::shape::*;
+
+const LABEL_OFFSET : f32 = 13.0;
+
+
+
+// ==============================================
+// === Utilities - Decimal Aligned Text Field ===
+// ==============================================
+
+/// Utility wrapper for a text field containing a float. Centers the string representation of the
+/// float on the decimal separator.
+mod decimal_aligned {
+    use super::*;
+
+    ensogl_core::define_endpoints! {
+        Input {
+            set_content(f32),
+        }
+        Output {}
+    }
+
+    impl component::Frp<Model> for Frp {
+        fn init(&self, app: &Application, model: &Model, _style: &StyleWatchFrp) {
+            let frp     = &self;
+            let network = &frp.network;
+            let _scene  = app.display.scene();
+
+            frp::extend! { network
+                formatted <- frp.set_content.map(|value| format!("{:.2}", value));
+                // FIXME: the next line is locale dependent. We need a way to get the current locale
+                //  dependent decimal separator for this.
+                //  See https://github.com/enso-org/ide/issues/1542 for progress on this.
+                left      <- formatted.map(|s| s.split('.').next().map(|s| s.to_string())).unwrap();
+
+                model.label_left.set_content  <+ left;
+                model.label_full.set_content       <+ formatted;
```

Comment:
spacing - here and in other places, please check all files (e.g the next line has also spacing issues)